### PR TITLE
ci: migrate to GitHub actions

### PR DESCRIPTION
**Pull request by FrankYang0529**
https://github.com/longhorn/longhorn/issues/8039

### CODE REVIEW

**Comment by a110605** [comment]
Hi @FrankYang0529 , shall we need to remove `ci-drone-PR` in https://github.com/longhorn/longhorn-ui/blob/16db32ebdf4fa513d2ed3d400aaee51a69f2e473/.github/mergify.yml#L4 ?

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    - name: Declare branch
+      run: |
+        echo "sha_short=$(git rev-parse --short "$GITHUB_SHA")" >> "$GITHUB_ENV"
+        echo "branch=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}" >> "$GITHUB_ENV"
+
+    - name: Login to Docker Hub
+      uses: docker/login-action@v3
+      with:
+        username: ${{ secrets.DOCKER_USERNAME }}
```
Comment (a110605) [inline .github/workflows/build.yml:33]: Just confirm, you will add secrets `DOCKER_USERNAME` and `DOCKER_PASSWORD` somewhere in github action, right?

Comment (FrankYang0529) [reply .github/workflows/build.yml:33]: Yes, already setup in organization level secrets.

**Comment by FrankYang0529** [comment]
> Hi @FrankYang0529 , shall we need to remove `continuous-integration/drone/pr ` in
> 
> https://github.com/longhorn/longhorn-ui/blob/16db32ebdf4fa513d2ed3d400aaee51a69f2e473/.github/mergify.yml#L4
> 
> ?

Yeah, we will do it after the PR is merged. Thanks.

**Review by a110605 — APPROVED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (a110605) [reply .github/workflows/build.yml:33]: thanks for confirmation.

**Review by innobead — APPROVED** [review]

**Comment by innobead** [comment]
@a110605 We need to add unit tests in the PR check in the future, also for Harvester.

**Comment by FrankYang0529** [comment]
@mergify backport v1.6.x

**Comment by FrankYang0529** [comment]
@mergify backport v1.5.x

**Pull request merged**